File: csy.py
from flask import Flask, jsonify, request, render_template

app = Flask(__name__)

tasks = []

@app.route('/')
def index():
    return render_template('index.html')

@app.route('/api/tasks', methods=['GET', 'POST'])
def task_list():
    if request.method == 'GET':
        return jsonify(tasks)
    elif request.method == 'POST':
        task = request.json['task']
        tasks.append(task)
        return jsonify({'message': 'Task added'})

if __name__ == '__main__':
    app.run(debug=True)

File: csy.js
function addTask() {
    var taskInput = document.getElementById('task');
    var taskValue = taskInput.value.trim();
  
    if (taskValue !== '') {
      fetch('/api/tasks', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json'
        },
        body: JSON.stringify({task: taskValue})
      })
      .then(response => response.json())
      .then(data => {
        var taskList = document.getElementById('taskList');
        var taskItemHTML = '';
  
        data.tasks.forEach(task => {
          taskItemHTML += '<li>' + task + '</li>';
        });
  
        taskList.innerHTML = taskItemHTML;
        taskInput.value = '';
      })
  